slither/detectors/erc/erc20/arbitrary_send_erc20.py

```diff
@@ -27,69 +59,57 @@ def no_permit_results(self) -> List[Node]:
-                else:
-                    ArbitrarySendErc20._arbitrary_from(f.nodes, self._no_permit_results)
-
     @staticmethod
-    def _arbitrary_from(nodes: List[Node], results: List[Node]):
+    def _arbitrary_from(ir: Operation, results: List[Node]):
         """Finds instances of (safe)transferFrom that do not use msg.sender or address(this) as from parameter."""
-        for node in nodes:
-            for ir in node.irs:
-                if (
-                    isinstance(ir, HighLevelCall)
-                    and isinstance(ir.function, Function)
-                    and ir.function.solidity_signature == "transferFrom(address,address,uint256)"
-                    and not (
-                        is_dependent(
-                            ir.arguments[0],
-                            SolidityVariableComposed("msg.sender"),
-                            node.function.contract,
-                        )
-                        or is_dependent(
-                            ir.arguments[0],
-                            SolidityVariable("this"),
-                            node.function.contract,
-                        )
-                    )
-                ):
-                    results.append(ir.node)
-                elif (
-                    isinstance(ir, LibraryCall)
-                    and ir.function.solidity_signature
-                    == "safeTransferFrom(address,address,address,uint256)"
-                    and not (
-                        is_dependent(
-                            ir.arguments[1],
-                            SolidityVariableComposed("msg.sender"),
-                            node.function.contract,
-                        )
-                        or is_dependent(
-                            ir.arguments[1],
-                            SolidityVariable("this"),
-                            node.function.contract,
-                        )
-                    )
+
+        if (
+            isinstance(ir, HighLevelCall)
+            and isinstance(ir.function, Function)
+            and ir.function.solidity_signature == "transferFrom(address,address,uint256)"
+            and ir.arguments[0]
+            not in (SolidityVariableComposed("msg.sender"), SolidityVariable("this"))
+        ):
+            results.append(ir.node)
+        elif (
+            isinstance(ir, LibraryCall)
+            and ir.function.solidity_signature
+            == "safeTransferFrom(address,address,address,uint256)"
+            and ir.arguments[1]
+            not in (SolidityVariableComposed("msg.sender"), SolidityVariable("this"))
+        ):
+            results.append(ir.node)
+
+    def _detect_arbitrary_from(self, instructions: List[Operation]):
+        has_permit = False
+        for ir in instructions:
+            if isinstance(ir, InternalCall):
+                self.simplify_and_analyze(ir.function.nodes)
```

Comment:
Mhh I think the recursive call here can lead to an infinite loop if the functions have a cycle.

Additionally I am a bit unsure about the strategy for internal call here. If you have an entry point that has an internal call, where the parameter is `msg.sender`, I think the logic here will fail to capture it
```solidity

function wrapper(address from, ..) internal{
   token.transferFrom(from, ...);
}

function externalFunction() external{
    wrapper(msg.sender, ...);
}

```

Because the functiond are analyzed independently here, I think it will lead to FP (unless I missed something?).